sklearn/cross_decomposition/pls_.py

```diff
@@ -293,6 +293,11 @@ def fit(self, X, Y):
             # 1) weights estimation (inner loop)
             # -----------------------------------
             if self.algorithm == "nipals":
+                # Replace columns that are all close to zero with zeros
+                Y_eps = np.finfo(Yk.dtype).eps
```

Comment:
`Y_eps = np.finfo(Yk.dtype).eps` should better be defined out of the for loop.

Reply:
Kind of makes me want to refactor all the `np.finfo(np.double).eps` calls out of this loop. 😅